compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -1363,10 +1365,11 @@ static LogicalResult setAttentionIntrinsicBasedVectorDistributionConfig(
   Value vMatrix = op.getValue();
 
   // Helper fn to store mma information.
-  auto storeMmaInfo = [](IREE::GPU::MmaInterfaceAttr mma,
+  auto storeMmaInfo = [](IREE::Codegen::InnerTileDescAttrInterface mma,
                          SmallVector<GPUIntrinsicType> &intrinsics) {
-    auto [mSize, nSize, kSize] = mma.getMNKShape();
-    auto [aType, bType, cType] = mma.getABCElementTypes();
+    auto [mSize, nSize, kSize] = IREE::GPU::MmaInterfaceAttr::getMNKShape(mma);
```

Comment:
I have suspicions that this'll explode.

I think what you'll want for all these is a utility function that does the right thing for MmaAttrInterface and ScaledMmaAttr and such and returns std::nullopt for anything unknown.

(and same comment on the below - I've been reviewing bottom-up)